libqtile/popup.py

```diff
@@ -67,6 +68,13 @@ def __init__(
         self.add_defaults(Popup.defaults)
         self.qtile = qtile
 
+        if "border" in config:
+            logger.warning(
+                'The "border" argument is obsolete and deprecated.',
+                'As a substitute, you may use "border_color" argument instead.',
+            )
```

Comment:
```suggestion
            logger.warning(
                'The "border" argument is obsolete and deprecated. '
                'As a substitute, you should use "border_color" argument instead.'
            )
```
The old `,` at the end of the first line meant you were passing two arguments to `logger.warning` when actually it should be a single string.

Also "may" -> "should".

This should be updated for the other messages.

Reply:
We should do tests too to check that the log messages are triggered when using deprecated configs.

An example is here:
https://github.com/qtile/qtile/blob/d827edd7b04c2fa95827c33aad3146794dfc39b2/test/widgets/test_launchbar.py#L31-L38

I don't think you'll need the monkeypatching, just initialise the widget with the deprecated parameters and check `caplog.records` for the messages.

I can help if you get stuck with these.